docs-website/topics/serialization-get-started.md

```diff
@@ -0,0 +1,232 @@
+
+> To set up the Kotlin compiler plugin for Bazel, see the example provided in the [rules_kotlin repository](https://github.com/bazelbuild/rules_kotlin/tree/master/examples/plugin/src/serialization).
+>
+{style="tip"}
+
+### Gradle
+
+Add the following dependency to your `build.gradle(.kts)` file:
+
+<tabs>
+<tab id="kotlin" title="Kotlin">
+
+```kotlin
+// build.gradle.kts
+plugins {
+    kotlin("plugin.serialization") version "%kotlinVersion%"
+}
+
+dependencies { 
+    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:%serializationVersion%")
+}
+```
+
+</tab>
+
+<tab id="groovy" title="Groovy">
+
+```groovy
+// build.gradle
+plugins {
+   id 'org.jetbrains.kotlin.plugin.serialization' version '%kotlinVersion%'  
+}
+
+dependencies {
+    implementation 'org.jetbrains.kotlinx:kotlinx-serialization-json:%serializationVersion%'
+}
+```
+
+</tab>
+
+</tabs>
+
+### Maven
+
+Add the serialization plugin and library to your `pom.xml` file:
+
+1. Specify the Kotlin and serialization version in the `<properties>` section:
+
+    ```xml
+    <properties>
+    <kotlin.version>%kotlinVersion%</kotlin.version>
+    <serialization.version>%serializationVersion%</serialization.version>
+    </properties>
+   ```
+
+2. Add the Kotlin serialization Maven plugin to the `<build>` section:
+
+   ```xml
+   <build>
+       <plugins>
+           <plugin>
+               <groupId>org.jetbrains.kotlin</groupId>
+               <artifactId>kotlin-maven-plugin</artifactId>
+               <version>${kotlin.version}</version>
+               <executions>
+                   <execution>
+                       <id>compile</id>
+                       <phase>compile</phase>
+                       <goals>
+                           <goal>compile</goal>
+                       </goals>
+                   </execution>
+               </executions>
+               <configuration>
+                   <compilerPlugins>
+                       <plugin>kotlinx-serialization</plugin>
+                   </compilerPlugins>
+               </configuration>
+               <dependencies>
+                   <dependency>
+                       <groupId>org.jetbrains.kotlin</groupId>
+                       <artifactId>kotlin-maven-serialization</artifactId>
+                       <version>${kotlin.version}</version>
+                   </dependency>
+               </dependencies>
+           </plugin>
+       </plugins>
+   </build>
+   ```
+
+3. Add the JSON serialization library dependency to the `<dependencies>` section:
+
+   ```xml
+   <dependencies>
+       <dependency>
+           <groupId>org.jetbrains.kotlinx</groupId>
+           <artifactId>kotlinx-serialization-json</artifactId>
+           <version>${serialization.version}</version>
+       </dependency>
+   </dependencies>
+   ```
+
+### Add the Kotlin serialization library to a multiplatform project
```

Comment:
We also mention Proguard rules (specific for Android) in the README.md. Do you plan to add them here?

Reply:
I think it's a bad idea. It is not always related to custom serializer, regular `@Serializable` classes can have named companions as well. I would expect separate page for Proguard then (linked here)